File: week0_explore/explore_architecture/03b_subagent_sdk/scripts/play.py
import os
import sys
import re
import argparse

# Add mud_manager to path
sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), "../../../mud_manager/lib")))
from mud_manager import Session

parser = argparse.ArgumentParser(description="MUD Client Script")
parser.add_argument("commands", nargs="+", help="Commands to send to the MUD server")
parser.add_argument("--user", default=os.environ.get("MUD_USER", "dummy"), help="MUD username")
parser.add_argument("--pass", dest="password", default=os.environ.get("MUD_PASS", "helloworld"), help="MUD password")
parser.add_argument("--host", default=os.environ.get("MUD_HOST", "localhost"), help="MUD host")
parser.add_argument("--port", type=int, default=int(os.environ.get("MUD_PORT", "4000")), help="MUD port")

args = parser.parse_args()

session = Session(host=args.host, port=args.port)
try:
    session.open()
    session.login(args.user, args.password)
    
    for cmd_str in args.commands:
        for sub_cmd in cmd_str.split(";"):
            sub_cmd = sub_cmd.strip()
            if not sub_cmd:
                continue
            
            session.send_command(sub_cmd)
            output = session.read_until_quiet(0.5)
            # Strip ANSI escape sequences
            clean_output = re.sub(r"\x1b\[[0-9;]*[A-Za-z]", "", str(output)).strip()
            print(clean_output)
            
except Exception as e:
    print(f"Error: {str(e)}", file=sys.stderr)
    sys.exit(1)
finally:
    if session.is_open():
        session.close()

File: week0_explore/explore_architecture/03b_subagent_sdk/scripts/play.rb
# frozen_string_literal: true

require_relative "../../../../../mud_manager/lib/mud_manager"

host = ENV.fetch("MUD_HOST", "localhost")
port = Integer(ENV.fetch("MUD_PORT", "4000"))
username = ENV.fetch("MUD_USER", "player")
password = ENV.fetch("MUD_PASS", "hellowworld")

if ARGV.empty?
  puts "Usage: ruby play.rb <command1> [command2] ..."
  exit 1
end

session = MudManager::Session.new(host: host, port: port)
begin
  session.open
  session.login(username, password)
  
  ARGV.each do |cmd_str|
    cmd_str.split(";").each do |sub_cmd|
      sub_cmd = sub_cmd.strip
      next if sub_cmd.empty?
      
      session.send_command(sub_cmd)
      output = session.read_until_quiet(0.5)
      clean_output = output.to_s.gsub(/\e\[[0-9;]*[A-Za-z]/, "").strip
      puts clean_output
    end
  end
rescue => e
  warn "Error: #{e.message}"
  exit 1
ensure
  session.close if session && session.open?
end
